feat(profile): improve profile customisation by adding posts/likes stats and tags

abaur3 opened the pull request:
# What Changes

This pull request improves the signed-in user's profile customisation by adding likes and posts stats, as well as tags describing the user's music taste.

## Key Implementations

### Likes and Posts Stats

- Extended Firestore profile documents with two new fields: `likes` (total number of likes received by the user’s samples) and `posts` (total number of samples posted by the user)
- Updated repository implementation (ProfileRepositoryFirebase): added field handling in `ensureProfile` (default to 0), deserialization in `toProfileOrNull` (default to 0 if missing).
- Extended `ProfileUiState` to include likes and posts count
- Connected the new fields in the `ProfileViewModel` so the UI reacts to profile updates.
- UI updates (view mode only): displayed “Likes” and “Posts” counters alongside followers/following stats.
- Rearranged the profile layout, placing the stats section just below the username.

### Tags Feature for Profile Customisation

- Extended Firestore profile documents with new field `tags`, consisting in a list of tags describing the user's music taste
- Included methods `addNewTag` and `removeTag` in `ProfileRepository` (and the corresponding implementation in `ProfileRepositoryFirebase` to handle tags operations
- Extended `ProfileUiState` to include tags, a string for handling tag insertion in edit mode and one for handling tag errors
- Connected the new fields in the `ProfileViewModel` so the UI reacts to profile updates.
- UI updates in both view and edit mode: in the edit mode, the user can add a new tag by typing it in the apposite field and clicking on the `+` button right next to it; this will add a chip for the tag (which will be shown right below the textField); moreover, each chip has its trailingIcon (the `x` button), making it easier for the user to delete the tag. In view mode, the tags are shown in view-only mode right below the bio.
-  Edit button is now pinned at the bottom of the screen, so that even when many tags are present and the view is scrollable, the button remains visible. 

## Notes

Posts/likes increasing and decreasing logic is not yet implemented. Current counters are just placeholders (same as followers/following count).

Closes #140 #139 #138 #142 #15

touunyy commented on app/src/main/java/com/neptune/neptune/ui/profile/ProfileScreen.kt:
```diff
@@ -107,10 +122,12 @@ fun ProfileScreen(
+    onRemoveTag: (String) -> Unit = {},
     onSettingsClick: () -> Unit = {},
     goBackClick: () -> Unit = {}
 ) {
```
You should find a way to have less parameters for your ProfileScreen (that was something Sonar told me once too). You can for example create a onClickFunctions data class that are going to regroup all your functions onClick

abaur3 commented:
Thank you for your review @touunyy !

I totally agree with you (and Sonar), right now the signature is really heavy with all these callbacks. I created a data class to include all onClick actions, and refactored the code to support this new structure (both in `ProfileScreen` and `ProfileScreenTest`).

I also fixed the text tag alignment in the `EditableTagChip` method: I replaced the `IconButton` with `Icon` to remove the 48 dp minimum size constraint that was inflating the chip height, and now the text is vertically centred. 

Don't hesitate to tell me if there's something else I should fix :)

AngeB21 commented on app/src/main/java/com/neptune/neptune/model/profile/Profile.kt:
```diff
@@ -12,6 +12,8 @@ package com.neptune.neptune.model.profile
  * @property subscriptions Number of other users this profile follows.
  * @property subscribers Number of users following this profile.
+ * @property likes Total number of likes received across all posts.
+ * @property posts Total number of posted samples.
```
You forgot the `@property` for tags

AngeB21 commented on app/src/main/java/com/neptune/neptune/ui/profile/ProfileScreen.kt:
The colors of the tags on the profile could follow the global theme of the app ( some light blue)

AngeB21 reviewed (changes requested):
Your code is very good!

Before approving the merge, I would like you to **normalize the tags when removing them** to ensure consistency and safer behavior.

As a side note, your method `ProfileEditContent` has a high Cognitive Complexity; you could consider breaking it down into smaller composables.
